ivy/functional/ivy/device.py

```diff
@@ -42,7 +42,7 @@
 dev_handles = dict()
 split_factors = dict()
 max_chunk_sizes = dict()
-
+soft_device_placement = True
```

Comment:
Let's have this as a stack `soft_device_mode_stack = list()` with getters and setters similar to the other stacks (e.g. `array_mode`, `shape_array_mode`, etc.)